src/PIL/MpoImagePlugin.py

```diff
@@ -92,7 +92,7 @@ def seek(self, frame):
             n = i16(self.fp.read(2)) - 2
             self.info["exif"] = ImageFile._safe_read(self.fp, n)
 
-            exif = self._getexif()
+            exif = self.getexif()
```

Comment:
The main difference is that `getexif` from `Image` class does not cache the parsed Exif tags.
So this call is done and forget when the user will need it.

This is not what we should expect.

Reply:
I'm not following, sorry. Yes, using `_getexif` would create a cache of the parsed exif tags, but it would also iterate over all the values.

From my understanding, the point of these PRs is to reduce operations on load to potentially increase speed - we don't need to work out all of the EXIF values when a user might not even be interested in them. When JpegImagePlugin and MpoImagePlugin only need two EXIF values each, isn't decoding all the values overkill?

Reply:
Sorry for the unclear review.
Yes, calling `getexif` is better, but right now, when reading an image, the `_getexif` function is called, and the result is cached so that if the user called it once again, the computation will not happen a second time.
This is not the case with `getexif`. If I trace a call to `Image.open("my_image.jpg").getexif()`, then the dictionary loading and some decoding will happens twice.

In my opinion, the `Exif` object should be stored into the `Image` instance once loaded. And if the user needs to modify it, this object could be cloned.

I hope it is clearer now :smile:

Reply:
Okay, see what you think of this idea - I have pushed a commit that changes `im.getexif()` so that it returns a shared instance of the Exif class, meaning that `getexif` decoding of EXIF values is now cached. This actually makes the `_getexif` "parsed_exif" cache largely redundant, because `_getexif` uses `getexif`, so I have removed that. Hopefully, that should make all scenarios faster.

Reply:
Very nice now :) Iterating 1000 times over an image (`open().getexif()`), I go from 18.0s to 8.11s. Confirmed by `%timeit` in IPython, going from 3.69ms per loop to 1.68ms per loop.

The last step (for me) is to get rid of `_fixup_dict` call of the tag 0x8769, but this should be in another PR.